src/ellipse.rs

```diff
@@ -0,0 +1,132 @@
+//! Implementation of ellipse shape.
+
+use std::f64::consts::PI;
+use std::{
+    iter,
+    ops::{Add, Sub},
+};
+
+use crate::{Arc, ArcAppendIter, PathEl, Point, Rect, Shape, Vec2};
+
+/// A ellipse.
+#[derive(Clone, Copy, Default, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Ellipse {
+    /// The center.
+    pub center: Point,
+    /// The radii.
+    pub radii: Vec2,
+    /// How much to rotate the major/minor axes.
+    pub x_rotation: f64,
+}
+
+impl Ellipse {
+    /// A new ellipse from center and radii.
```

Comment:
and x_rotation?

It would be nice if we could include, in the docs, a link to some resource (wikipedia, mathematica) that explains how this representation works (or if we just do it ourselves)?